### ISSUE TITLE
onError name printing

### ISSUE THREAD

**Issue by hochgi** [scallop/scallop#30]
in current state, when an error occurs, an error message is printed which says:
    `[scallop] Error: some error...`

i wanted to change that behavior, and it resulted with re-writing everything you already written in the `onError` method, within my `Conf` object definition:   

```
...
private[Conf] object throwError extends scala.util.DynamicVariable[Boolean](false)

override def onError(e: Throwable): Unit = e match {
    case r: ScallopResult if !throwError.value => r match {
        case Help =>
            builder.printHelp
            sys.exit(0)
    case Version =>
        builder.vers.foreach(println)
        sys.exit(0)
case ScallopException(message) =>
        if (System.console() == null) {
            // no colors on output
            println("[My Very Cool App Name] Error: %s" format message)
        } else {
            println("[\033[31mMy Very Cool App Name\033[0m] Error: %s" format message)
        }
        sys.exit(1)
    }
    case e => throw e
}
...
```

it would be great to have a nicer way of doing that.

**Comment by Rogach** [comment]
You only need to change `[scallop]` part?

**Comment by Rogach** [comment]
One note - that `throwError` stuff is needed to make testing of the library easier, you probably don't need to rewrite that.

**Comment by hochgi** [comment]
yep. just the "scallop" name... maybe someone else might want to reformat the error message, or maybe even print it to a logger instead of stdout, so it might be better to get a print function instead of string. something like `errorHandler: (errorMsg: String) => ()` which you can invoke instead of `println`

**Comment by Rogach** [comment]
I'll add two overridable values on `ScallopConf` - one to specify the 
printed name (`[scallop]`), and another (errorMessageHandler) - to give 
more flexibility in error formatting/rerouting.

**Comment by hochgi** [comment]
thanks! it would be awesome!

**Comment by Rogach** [comment]
I published new version (`0.6.4`) with this change. It's already published on Sonatype (so if you are using sbt, you can switch to it right away) and in 1-2 hours it would sync to Maven Central.

**Comment by hochgi** [comment]
i have switched to the new version.
thanks!!!
